static/js/on-off-switch.js

```diff
@@ -77,23 +77,36 @@ OnOffSwitch.prototype.updateStatus = function() {
       'Accept': 'application/json'
     }
   };
-  fetch(this.onPropertyUrl, opts).then((function(response) {
+
+  fetch(this.onPropertyUrl, opts).then(function(response) {
     return response.json();
-  }).bind(this)).then((function(response) {
-    this.properties.on = response.on;
-    if (response.on === null) {
-      return;
-    } else if(response.on) {
-      this.showOn();
-    } else {
-      this.showOff();
-    }
+  }).then((function(response) {
+    this.onPropertyStatus(response);
   }).bind(this)).catch(function(error) {
     console.error('Error fetching on/off switch status ' + error);
   });
 };
 
 /**
+ * Handle a 'propertyStatus' message
+ * @param {Object} properties - property data
+ */
+OnOffSwitch.prototype.onPropertyStatus = function(data) {
+  if (!data.hasOwnProperty('on')) {
+    return;
+  }
+  this.properties.on = data.on;
+  if (data.on === null) {
+    return;
+  }
+  if (data.on) {
+    this.showOn();
+  } else {
+    this.showOff();
+  }
```

Comment:
Should this code and the code for drawing it in the first place be factored?

Perhaps have a show function which is called from both places?

Reply:
The code for drawing the initial state of the OnOffSwitch uses `onPropertyStatus` as well [here](https://github.com/mozilla-iot/gateway/pull/281/files#diff-5343f48079b432eb1ed84e1f67cdc038R84) then it's updated in response to messages [here](https://github.com/mozilla-iot/gateway/pull/281/files#diff-3760c5a1e06ccd33ec83b97e6a996154R41)

Reply:
I was probably looking at the unpatched code and missed that it was changed in this PR :)